frontend/apps/reader/modules/readerbookmark.lua

```diff
@@ -884,7 +902,7 @@ end
 
 function ReaderBookmark:getBookmarkIndexFullScan(item)
     for i, v in ipairs(self.bookmarks) do
-        if item.datetime == v.datetime and item.page == v.page then
+        if item.datetime == v.datetime then
```

Comment:
Why did we check `page` before? And why no longer needed?

Reply:
(1) Tha main purpose is to speed up the bookmarks as much as possible. For that, I moved fetching highlight drawer from the start to filtering by highlight style.
(2) I belive comparing `datetime` is enough to find the corresponding bm/hl.
Actually, we do it in half of the cases:
https://github.com/koreader/koreader/blob/72a6fa1e641f1a380f33c62cc47744334ed82798/frontend/apps/reader/modules/readerbookmark.lua#L1453
https://github.com/koreader/koreader/blob/72a6fa1e641f1a380f33c62cc47744334ed82798/frontend/apps/reader/modules/readerbookmark.lua#L1466
https://github.com/koreader/koreader/blob/72a6fa1e641f1a380f33c62cc47744334ed82798/frontend/apps/reader/modules/readerhighlight.lua#L1890
https://github.com/koreader/koreader/blob/72a6fa1e641f1a380f33c62cc47744334ed82798/frontend/apps/reader/modules/readerhighlight.lua#L2155
and that causes no complaints.